librz/core/cbin.c

```diff
@@ -2790,15 +2817,38 @@ static int bin_sections(RzCore *r, PJ *pj, int mode, ut64 laddr, int va, ut64 at
 		fd = rz_core_file_cur_fd(r);
 		rz_flag_space_set(r->flags, print_segments ? RZ_FLAGS_FS_SEGMENTS : RZ_FLAGS_FS_SECTIONS);
 	}
-	if (IS_MODE_NORMAL(mode)) {
+	if (IS_MODE_NORMAL(mode) && !print_segments ) {
+		//rz_cons_println("");
+		if(!strncmp(r->file->core->bin->cur->o->info->rclass, "elf", 3) ||  !strncmp(r->file->core->bin->cur->o->info->rclass, "mach", 4)){
```

Comment:
Is this big if statement to check if the current plugin defined types and flags? If so could this be replaced by a
```c
if(plugin && (plugin->section_type_to_string || plugin->section_flags_to_rzlist))
``` 
?